src/store/category.js

```diff
@@ -0,0 +1,27 @@
+import api from '../plugins/axios';
+
+const state = () => ({
+  category: []
+});
+
+const getters = {
+  allCategory: state => state.category
+};
+
+const mutations = {
+  setCategory: (state, data) => (state.category = data)
+};
+
+const actions = {
+  // get CATEGORY data from API
+  fetchCategory({ commit }) {
+    api
+      .get('/category')
+      .then(res => {
+        commit('setCategory', res.data);
```

Comment:
we should be check response data be for do any action to make sure properties of object exist
if (res) {
}